examples/contrib/oed/models/bayes_linear.py

```diff
@@ -1,15 +1,126 @@
+    return model
+
+
+def group_normal_guide(observation_sd, coef1_shape, coef2_shape,
+                       coef1_label="w1", coef2_label="w2"):
+    return partial(
+        normal_inv_gamma_family_guide, w_sizes={coef1_label: coef1_shape, coef2_label: coef2_shape},
+        obs_sd=observation_sd)
+
+
+def zero_mean_unit_obs_sd_lm(coef_sd, coef_label="w"):
+    model = known_covariance_linear_model(torch.tensor(0.), coef_sd, torch.tensor(1.), coef_label=coef_label)
+    guide = normal_guide(torch.tensor(1.), coef_sd.shape, coef_label=coef_label)
+    return model, guide
+
+
+def normal_inverse_gamma_linear_model(coef_mean, coef_sqrtlambda, alpha,
+                                      beta, coef_label="w",
+                                      observation_label="y"):
+    return partial(bayesian_linear_model,
+                   w_means={coef_label: coef_mean},
+                   w_sqrtlambdas={coef_label: coef_sqrtlambda},
+                   alpha_0=alpha, beta_0=beta,
+                   response_label=observation_label)
+
+
+def normal_inverse_gamma_guide(coef_shape, coef_label="w", **kwargs):
+    return partial(normal_inv_gamma_family_guide, obs_sd=None, w_sizes={coef_label: coef_shape}, **kwargs)
+
+
+def logistic_regression_model(coef_mean, coef_sd, coef_label="w", observation_label="y"):
+    return partial(bayesian_linear_model,
+                   w_means={coef_label: coef_mean},
+                   w_sqrtlambdas={coef_label: 1./coef_sd},
+                   obs_sd=torch.tensor(1.),
+                   response="bernoulli",
+                   response_label=observation_label)
+
+
+def lmer_model(fixed_effects_sd, n_groups, random_effects_alpha, random_effects_beta,
+               fixed_effects_label="w", random_effects_label="u", observation_label="y",
+               response="normal"):
+    return partial(bayesian_linear_model,
+                   w_means={fixed_effects_label: torch.tensor(0.)},
+                   w_sqrtlambdas={fixed_effects_label: 1./fixed_effects_sd},
+                   obs_sd=torch.tensor(1.),
+                   re_group_sizes={random_effects_label: n_groups},
+                   re_alphas={random_effects_label: random_effects_alpha},
+                   re_betas={random_effects_label: random_effects_beta},
+                   response=response,
+                   response_label=observation_label)
+
+
+def sigmoid_model(coef1_mean, coef1_sd, coef2_mean, coef2_sd, observation_sd,
+                  sigmoid_alpha, sigmoid_beta, sigmoid_design,
+                  coef1_label="w1", coef2_label="w2", observation_label="y",
+                  sigmoid_label="k"):
+
+    def model(design):
+        batch_shape = design.shape[:-2]
+        k_shape = batch_shape + (sigmoid_design.shape[-1],)
+        k = pyro.sample(sigmoid_label,
+                        dist.Gamma(sigmoid_alpha.expand(k_shape),
+                                   sigmoid_beta.expand(k_shape)).independent(1))
```

Comment:
In this sigmoid model, the random slope has a Gamma distribution. Could easily add a different one and use `bayesian_linear_model` to do the rest